docs/source/traits_user_manual/advanced.rst

```diff
@@ -301,6 +301,38 @@ exception, as does attempting to set an attribute that is not one of the three
 defined attributes. In essence, TreeNode behaves like a type-checked data
 structure.
 
+.. index:: HasRequiredTraits class
+
+.. _hasrequiredtraits:
+
+HasRequiredTraits
+'''''''''''''''''
+
+This class builds on the functionality of HasStrictTraits and ensures
+that any object attribute with `required=True` in its metadata must be passed
+as an argument on object initialization.
+
+An example of a class with required traits::
+
+    class RequiredTest(HasRequiredTraits):
+        required_trait = Any(required=True)
+        non_required_trait = Any()
+
+All required traits have to be provided as arguments on creating a new
+instance::
+
+   >>> new_instance = RequiredTest(required_trait=13.0)
```

Comment:
Please can you fix up the indentation to be consistent? (This code block is indented 3 spaces; the one above uses 4 spaces, and the ones below are 2 spaces. Pick one!)

Reply:
FWIW, most of the rest of this document seems to use 4 spaces, so I'd go with that.

Reply:
Looks like I need to sort out my tabs/spaces configs in atom!

Reply:
Yeah, reST is weird, though. It encourages *3-space* indents, so you tend to find a mix of 3-space and 4-space indents actual reST documentation.